core-bundle/src/Resources/contao/widgets/Picker.php

```diff
@@ -174,11 +174,18 @@ public function generate()
 
 	protected function generateValues($blnHasOrder): array
 	{
-		$strRelatedTable = $this->getRelatedTable();
+		if (substr($this->context ?? '', 0, 3) === 'dc.')
+		{
+			$strRelatedTable = substr($this->context, 3);
+		}
+		else
+		{
+			$strRelatedTable = $this->getRelatedTable();
+		}
 
 		if (!$strRelatedTable)
 		{
-			return (array) $this->varValue;
+			return array_combine((array) $this->varValue, (array) $this->varValue);
```

Comment:
I’m not sure if this change is correct. But the widget seems to be expecting the values as keys. Without this change a selected value does not get stored because the key is `0` instead of the actual selected ID value.

Reply:
If `$this->varValue` is an array of IDs, you are right. Not sure for UUIDs though.

Reply:
Looking at like 121, I’m pretty sure the array keys are meant to be the “raw values” and the array values are the labels shown to the user.